retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp

```diff
@@ -568,11 +568,37 @@ void CreateGxsForumMsg::fileHashingFinished(QList<HashedFile> hashedFiles)
 	QList<HashedFile>::iterator it;
 	for (it = hashedFiles.begin(); it != hashedFiles.end(); ++it) {
 		HashedFile& hashedFile = *it;
+		QString ext = QFileInfo(hashedFile.filename).suffix().toUpper();
+
 		RetroShareLink link = RetroShareLink::createFile(hashedFile.filename, hashedFile.size,
 		                                                 QString::fromStdString(hashedFile.hash.toStdString()));
-		if (link.valid()) {
-			mesgString += link.toHtmlSize() + "<br>";
+
+		bool preview = false;
+		if(hashedFiles.size()==1 && (ext == "JPG" || ext == "PNG" || ext == "JPEG" || ext == "GIF" || ext == "WEBP"))
+		{
+			QString encodedImage;
+			uint32_t maxImageSize = 32000;
+			if (RsHtml::makeEmbeddedImage(hashedFile.filepath, encodedImage, 320*240, maxImageSize - 200 - link.toHtmlSize().length()))
```

Comment:
Check link valid upper as you use it here.

Reply:
i dont understand what you mean,
i has only cloned code from chatwidget, there is no link valid check
`void ChatWidget::fileHashingFinished(QList<HashedFile> hashedFiles)
`
where you want the `if (link.valid())` check?